src/arraymancer/laser/tensor/initialization.nim

```diff
@@ -243,23 +243,31 @@ proc fromBuffer*[T](rawBuffer: ptr UncheckedArray[T], shape: varargs[int]): Tens
   ##
   ## Its counterpart ``toUnsafeView`` can be used to obtain ``ptr UncheckedArray`` from a Tensor.
   var size: int
-  initTensorMetadata(result, size, shape)
+  initTensorMetadata(result, size, shape, layout)
   cpuStorageFromBuffer(result.storage, rawBuffer, size)
 
-proc fromBuffer*[T](rawBuffer: pointer, shape: varargs[int]): Tensor[T] =
+proc fromBuffer*[T](rawBuffer: ptr UncheckedArray[T], shape: varargs[int]): Tensor[T] =
+  ## Call `fromBuffer` with layout = rowMajor
+  fromBuffer[T](rawBuffer, shape, rowMajor)
+
+proc fromBuffer*[T](rawBuffer: pointer, shape: varargs[int], layout: static OrderType): Tensor[T] =
   ## Creates a `Tensor[T]` from a raw `pointer`. Make sure that the explicit type
   ## given to this proc actually matches the data stored behind the pointer!
   ## The size derived from the given shape must match the size of the buffer!
   ##
   ## Its counterpart ``toUnsafeView`` can be used to obtain ``ptr UncheckedArray`` from a Tensor.
   var size: int
-  initTensorMetadata(result, size, shape)
+  initTensorMetadata(result, size, shape, layout)
   cpuStorageFromBuffer(result.storage, rawBuffer, size)
 
-func toUnsafeView*[T: KnownSupportsCopyMem](t: Tensor[T], aligned: static bool = true): ptr UncheckedArray[T] {.inline.} =
+proc fromBuffer*[T](rawBuffer: pointer, shape: varargs[int]): Tensor[T] =
+  ## Call `fromBuffer` with layout = rowMajor
+  fromBuffer[T](rawBuffer, shape, rowMajor)
+
+func toUnsafeView*[T: KnownSupportsCopyMem](t: Tensor[T], offset: int = 0, aligned: static bool = true): ptr UncheckedArray[T] {.inline.} =
```

Comment:
Use-case of this offset?

Reply:
Just a cleaner way to do ``unsafeAddr(toUnsafeView(t, x)[offset])``

Reply:
But is there a need for this construct for external libraries? `unsafeAddr(toUnsafeView(t)[offset])`

Reply:
It's for using Tensor as a high level buffer construct that gets passed to a C API (thus the ptr API). Each call of the C API uses a different part of the buffer to store sensor data. 

When hw operation is done, a simple reshape and you get sensor data ordered easily accessible. 

Might be a marginal use case but I still find it handy and just cleaner than calling an unnecessary unsafeAddr on top of unsafeView - but it's not strictly necessary so we can remove it if you wish.

Reply:
@mratsim So in the end, what's the consensus ? Do you want me to remove offset from the API ?

Reply:
I think it's best to remove it for now.

Once a tensor is provided as a raw buffer, the end user takes complete ownership and can write their own glue code. I'd rather keep the raw/unsafe API as simple as possible.